packages/uicore/src/components/TagInput/TagInput.tsx

```diff
@@ -274,7 +264,7 @@ export function TagInput<T>(props: TagInputProps<T>) {
         return (
           <MenuItem
             key={keyOf(item)}
-            text={labelFor(getLabel(item))}
```

Comment:
`getLabel` is not needed as the return value of `labelFor` should be used directly without modifying it further internally.